base/strings/search.jl

```diff
@@ -174,9 +194,12 @@ function _searchindex(s::String, t::String, i::Integer)
     _searchindex(unsafe_wrap(Vector{UInt8},s), unsafe_wrap(Vector{UInt8},t), i)
 end
 
-function _searchindex(s::ByteArray, t::ByteArray, i::Integer)
-    n = sizeof(t)
-    m = sizeof(s)
+function _searchindex(s::AbstractVector{T},
+                      t::AbstractVector{T},
```

Comment:
Probably `AbstractVector{<:Union{Int8,UInt8}}`, so that `s` and `t` don't have to have the same element type.